cmake/FindAutotalks.cmake

```diff
@@ -0,0 +1,78 @@
+        PATHS $ENV{HOME}
+        PATH_SUFFIXES "autotalks_craton_api"
+        CMAKE_FIND_ROOT_PATH_BOTH # Seems like this is not needed in newer CMake
+    )
+
+    set(AUTOTALKS_INCLUDE_DIRS "${AUTOTALKS_ROOT}/include" "${AUTOTALKS_ROOT}/ref_src/include" "${AUTOTALKS_ROOT}/depend/device/include" "${AUTOTALKS_ROOT}/src/include" "${AUTOTALKS_ROOT}/src/core/include")
+
+    set(AUTOTALKS_LIBS_DIR ${AUTOTALKS_ROOT}/output.cr2_lnx/armv7-32/lib)
+    set(AUTOTALKS_OBJ_DIR ${AUTOTALKS_ROOT}/output.cr2_lnx/armv7-32/obj)
+
+    if (NOT TARGET Autotalks::ToMcrypt)
+        add_library(Autotalks::ToMcrypt UNKNOWN IMPORTED)
+        set_target_properties(Autotalks::ToMcrypt PROPERTIES
+            IMPORTED_LOCATION "${AUTOTALKS_LIBS_DIR}/libtomcrypt.a"
+            INTERFACE_INCLUDE_DIRECTORIES "${AUTOTALKS_INCLUDE_DIRS}")
+    endif()
+
+    if (NOT TARGET Autotalks::ToMmath)
+        add_library(Autotalks::ToMmath UNKNOWN IMPORTED)
+        set_target_properties(Autotalks::ToMmath PROPERTIES
+            IMPORTED_LOCATION "${AUTOTALKS_LIBS_DIR}/libtommath.a"
+            INTERFACE_INCLUDE_DIRECTORIES "${AUTOTALKS_INCLUDE_DIRS}")
+    endif()
+
+    if (NOT TARGET Autotalks::Optee)
+        add_library(Autotalks::Optee UNKNOWN IMPORTED)
+        set_target_properties(Autotalks::Optee PROPERTIES
+            IMPORTED_LOCATION "${AUTOTALKS_ROOT}/depend/optee/lib/libteec.so"
+            INTERFACE_INCLUDE_DIRECTORIES "${AUTOTALKS_INCLUDE_DIRS}")
+    endif()
+
+    set(AUTOTALKS_OBJECTS ${AUTOTALKS_OBJ_DIR}/target_type/target_type_local.o ${AUTOTALKS_OBJ_DIR}/platform_type/platform_type_linux_u.o ${AUTOTALKS_OBJ_DIR}/link_layer/linux_u/local/link_layer_local_shared_memory.o ${AUTOTALKS_OBJ_DIR}/link_layer/linux_u/ipc/link_layer_ipc.o ${AUTOTALKS_OBJ_DIR}/link_layer/linux_u/tee/static_list.o ${AUTOTALKS_OBJ_DIR}/link_layer/linux_u/tee/link_layer_tee.o ${AUTOTALKS_OBJ_DIR}/link_layer/link_layer_interface.o ${AUTOTALKS_OBJ_DIR}/flash_reader/linux_u/local/flash_reader_local.o ${AUTOTALKS_OBJ_DIR}/ref_sys/ref_sys.o ${AUTOTALKS_OBJ_DIR}/ref_sys/ref_sys_logger.o ${AUTOTALKS_OBJ_DIR}/log_reader/log_reader.o ${AUTOTALKS_OBJ_DIR}/secure_hdif_crypto_layer/crypto_layer_stubs.o ${AUTOTALKS_OBJ_DIR}/time_sync/time_sync.o ${AUTOTALKS_OBJ_DIR}/time_sync/time_sync_poti.o ${AUTOTALKS_OBJ_DIR}/poti_lib/gnss/gnss.o ${AUTOTALKS_OBJ_DIR}/poti_lib/nmea/nmea.o ${AUTOTALKS_OBJ_DIR}/poti_lib/nmea/nmea_io.o ${AUTOTALKS_OBJ_DIR}/poti_lib/nmea/nmea_io_check.o ${AUTOTALKS_OBJ_DIR}/poti_lib/nmea/nmea_nav.o ${AUTOTALKS_OBJ_DIR}/poti_lib/nmea/nmea_parse.o ${AUTOTALKS_OBJ_DIR}/poti_lib/gnss/teseo/teseo.o ${AUTOTALKS_OBJ_DIR}/poti_lib/gnss/teseo/teseo_commands.o ${AUTOTALKS_OBJ_DIR}/poti_lib/coord/coord.o ${AUTOTALKS_OBJ_DIR}/poti_lib/periodic_alarm/periodic_alarm.o ${AUTOTALKS_OBJ_DIR}/poti_lib/api/poti_api.o ${AUTOTALKS_OBJ_DIR}/poti_lib/ubx/serial_driver.o ${AUTOTALKS_OBJ_DIR}/poti_lib/ubx/ubx_driver.o ${AUTOTALKS_OBJ_DIR}/poti_lib/ubx/ubx_nav.o)
```

Comment:
You can improve readability by using `list(TRANSFORM AUTOTALKS_OBJECTS PREPEND ${AUTOTALKS_OBJ_DIR}/)` and omit the repeated prefix in the `set` call.